File: ruby/7kyu/dna.rb
$map = {"A" => "T", "T" => "A", "C" => "G", "G" => "C"} # global variable

def DNA_strand(dna)
	strip = ""
	dna.each_char{ |s|
		strip += $map[s] # to access it must use '$'
	}
	strip
end 

dna = "ATTGC"

puts DNA_strand(dna)

File: c++/7kyu/dna.cpp
#include <string>
#include <iostream>
#include <map>

using namespace std;
std::string DNAStrand(const std::string& dna)
{
  map<char, char> d {{'A', 'T'}, {'T', 'A'}, {'C', 'G'}, {'G', 'C'}};
  string side;
  for(char c: dna){
    side += d[c];
  }
  return side;
}